src/Expressions/FunctionExpression.php

```diff
@@ -106,9 +106,8 @@ private function sqlCount(row $rows, AsyncMysqlConnection $conn): int {
       foreach ($rows as $row) {
         $row as dict<_, _>;
         $val = $expr->evaluate(/* HH_FIXME[4110] generics */ $row, $conn);
-        if (!C\contains_key($buckets, $val)) {
-          $buckets[$val] = 1;
-        }
+        // Warning for future reader. This expression assigns to a `mixed` key.
+        $buckets[$val] = 1;
```

Comment:
This loop checked for the existence of a key in each iteration. However it would always write a constant 1 integer literal. The result was giving each key a value of one. This looks like a useless if to me. Please correct me if I am wrong.